src/db/utils.js

```diff
@@ -6,44 +6,38 @@ const createCustomDbError = message => {
 
 // static method for Mongoose schema
 const findByIdOrError = function (id) {
-  return new Promise((resolve, reject) => {
-    this.findById(id)
-      .then(user => {
-        if (!user) {
-          return reject(createCustomDbError('No document matching that id'))
-        }
-        resolve(user)
-      })
-      .catch(reject)
-  })
+  return this.findById(id)
+    .then(res => {
+      if (!res) {
+        return Promise.reject(createCustomDbError('No document matching that id'))
+      }
+      return Promise.resolve(res)
+    })
+    .catch(err => Promise.reject(err))
```

Comment:
like: I am gonna catch a rejected promise just so I can reject it again

Just seem mean to the error